### PR TITLE
Chet 169 db config parse

### PR DESCRIPTION
**Pull request by theghostwhoforks**
Fixes bean load on startup with a h2 database

### CODE REVIEW

**Comment by theghostwhoforks** [comment]
The same code is present in `CHET-169-db-state-machine`. I pulled it here to allow for master to be pristine

File: src/main/java/com/atlassian/migration/datacenter/core/db/DatabaseExtractorFactory.java [hunk 1]
```diff
@@ -19,16 +19,20 @@
...
     public static DatabaseExtractor getExtractor(ApplicationConfiguration config) throws DatabaseMigrationFailure
     {
         if (config.getDatabaseConfiguration().getType().equals(DBType.POSTGRESQL)) {
             return new PostgresExtractor(config);
-        } else {
-            throw new DatabaseMigrationFailure("Unsupported database type: " + config.getDatabaseConfiguration().getType());
         }
+
+        //Profile scoped perhaps?
+        if (config.getDatabaseConfiguration().getType().equals(DBType.H2)) {
```
Comment (nanux) [inline src/main/java/com/atlassian/migration/datacenter/core/db/DatabaseExtractorFactory.java]: With H2, I don't think we need to fail directly. We can just skip the dump and copy the database when we are running the filesystem transfer (the second one). I am not really fussed about it as H2 is not really a target platform so complicating the code is not necessary.

Comment (bpar476) [reply src/main/java/com/atlassian/migration/datacenter/core/db/DatabaseExtractorFactory.java]: I think I agree as testing locally with H2 makes it quite fast. However, we may want to introduce a run profile which starts jira with postgres in docker

Comment (theghostwhoforks) [reply src/main/java/com/atlassian/migration/datacenter/core/db/DatabaseExtractorFactory.java]: Yeah - we need to start in postgres by default going forward. Once we create a run profile, this is not necessary

Comment (theghostwhoforks) [reply src/main/java/com/atlassian/migration/datacenter/core/db/DatabaseExtractorFactory.java]: For now, this is necessary to start the plugin. Without this, the app start will fail as the database configuration throws an exception

**Pull request merged**